blocks/extensions/training.py

```diff
@@ -0,0 +1,28 @@
+from blocks.extensions import TrainingExtension
+
+
+class AdjustParameter(TrainingExtension):
+    """Adjusts shared variable parameter using some function.
+
+    Applies function to compute the new value of a shared parameter each
+    iteration.
+
+    This class can be used to adapt over the training process parameters like
+    learning rate, momentum, etc.
+
+    Parameters
+    ----------
+    parameter : shared variable to be adjusted
+    function : function which input number of aggregated examples and outputs
+               the parameter value
+
+    """
+    def __init__(self, parameter, function, **kwargs):
+        super(AdjustParameter, self).__init__(**kwargs)
+        self.learning_rate = parameter
```

Comment:
If you want this class to be generic enough for updating any shared variable (which I think is a good idea), then this attribute should probably be `self.parameter` and not `self.learning_rate`.